### PR TITLE
add llama_get_pooling_type function

### PR DESCRIPTION
**Pull request by iamlemec**
This allows the user to inspect the realized value of the context's `pooling_type`. In cases where `UNSPECIFIED` is passed as the context parameter, this helps identify the default pooling type coming from the model. This is useful because, depending on the pooling type, different values for `logits` in the batch may need to be passed and the embeddings outputs are of different size (and require different functions to access).

### CODE REVIEW

File: llama.h [hunk 1]
```diff
@@ -390,8 +390,9 @@ extern "C" {
     LLAMA_API uint32_t llama_n_ubatch   (const struct llama_context * ctx);
     LLAMA_API uint32_t llama_n_seq_max  (const struct llama_context * ctx);
 
-    LLAMA_API enum llama_vocab_type llama_vocab_type(const struct llama_model * model);
-    LLAMA_API enum llama_rope_type  llama_rope_type (const struct llama_model * model);
+    LLAMA_API enum llama_vocab_type   llama_vocab_type  (const struct llama_model   * model);
+    LLAMA_API enum llama_rope_type    llama_rope_type   (const struct llama_model   * model);
+    LLAMA_API enum llama_pooling_type llama_pooling_type(const struct llama_context * model);
```
Comment (ggerganov) [inline llama.h:395]: Move up next to the other `llama_context` functions

```suggestion
    LLAMA_API enum llama_pooling_type llama_pooling_type(const struct llama_context * ctx);
```

**Review by ggerganov — APPROVED** [review]

File: llama.h (hunk 1 above)
Comment (iamlemec) [reply llama.h]: Fixed the argument name. Moved up higher, but didn't seem like aligning signature was worth it.
